### ISSUE TITLE
hi,你的simply_ppo代码中的几个错误：

### ISSUE THREAD

**Issue by clicdl** [MorvanZhou/Reinforcement-learning-with-tensorflow#122]
1. critic 网络用来计算出v, LOSS函数应该类似dqn用迭代v的方式去逼近真实的v，再用A=Q-V的方式求出A。 而你的loss函数是去最小化advantage，怎么可以最小化advantage，我们的目标就是要尽量增大优势函数的。
 # critic
        with tf.variable_scope('critic'):
            l1 = tf.layers.dense(self.tfs, 100, tf.nn.relu)
            self.v = tf.layers.dense(l1, 1)
            self.tfdc_r = tf.placeholder(tf.float32, [None, 1], 'discounted_r')
            self.advantage = self.tfdc_r - self.v
            self.closs = tf.reduce_mean(tf.square(self.advantage))
            self.ctrain_op = tf.train.AdamOptimizer(C_LR).minimize(self.closs)
2. 整个代码中只在update函数中去run train_op, 而这个函数第一步就是去把old_pi的参数更新了，致使你计算的每个pi和oldpi都是一样的东西。不符合ppo算法的公式，ppo算法应当是多步去计算pi，而oldpi不变，n步后才去更新一次old_pi。
  def update(self, s, a, r):
        self.sess.run(self.update_oldpi_op)
        adv = self.sess.run(self.advantage, {self.tfs: s, self.tfdc_r: r})
        # adv = (adv - adv.mean())/(adv.std()+1e-6)     # sometimes helpful

        # update actor
        if METHOD['name'] == 'kl_pen':
            for _ in range(A_UPDATE_STEPS):
                _, kl = self.sess.run(
                    [self.atrain_op, self.kl_mean],
                    {self.tfs: s, self.tfa: a, self.tfadv: adv, self.tflam: METHOD['lam']})

**Comment by MorvanZhou** [comment]
你好，
1. critic 的任务是最小化对奖励的估计误差，也就是这里的adv。
2. 不管是把update_oldpi_op 放在n步更新前面还是n步更新后面，达到的目的是一样的。只要在n步更新过程中，oldpi是fixed就行。
